rescript

```diff
@@ -498,6 +498,18 @@ if (
         reasons_to_rebuild = [];
         LAST_BUILD_START = +Date.now();
       }
+      // if acquiring lock failed, no need retry here
+      // since build_finished_callback will try again
+      // however this is no longer the case for multiple-process
+      // it could fail due to other issues like .bsb.lock
+      else {
+        dlog(`Acquire lock failed, do the build later ${depth} : ${reasons_to_rebuild}`);
+        var waitTime = Math.pow(2, depth) * 40;
+        setTimeout(function () {
+          var d = Math.min(depth + 1, 5);
+          build(d);
+        }, waitTime);
+      }
     }
```

Comment:
If we only lock using `is_build` this is not needed. since any reasons to build left will be triggered by `build_finished_callback` later.
However, due to file locking competition against another `rescript build`, we failed to acquire the lock, we should `retry` here